app/models/manageiq/providers/redhat/infra_manager/refresh/parse/strategies/vm_inventory.rb

```diff
@@ -153,27 +153,58 @@ def vm_inv_to_guest_device_hashes(inv, lan_uids)
     end
 
     def vm_inv_to_network_hashes(inv, guest_device_uids)
-      reported_device = inv.respond_to?(:reported_devices) ? inv.reported_devices[0] : nil
-      inv_net = reported_device.ips if reported_device
-      result = []
-      return result if inv_net.nil?
+      reported_devices = inv.respond_to?(:reported_devices) ? inv.reported_devices : nil
+      all_networks = []
+      return all_networks if !reported_devices || reported_devices.blank?
+
+      network_by_mac = {}
+      host_name = inv.fqdn if inv.respond_to?(:fqdn)
+      reported_devices.to_miq_a.each do |reported_device|
+        inv_net = reported_device.ips
+        next if inv_net.nil?
 
-      inv_net.to_miq_a.each do |data|
-        new_result = { :ipaddress => data.address }
-        result << new_result unless new_result.blank?
+        ipaddresses = get_ipaddresses(inv_net)
+
+        ipaddresses.each do |ipv4address, ipv6address|
+          new_network = {
+            :hostname => host_name
+          }
+          new_network[:ipaddress] = ipv4address unless ipv4address.nil?
+          new_network[:ipv6address] = ipv6address unless ipv6address.nil?
+
+          device_mac = reported_device.mac ? reported_device.mac.address : nil
+          network_by_mac[device_mac] = new_network if device_mac && !network_by_mac[device_mac]
+          all_networks << new_network
+        end
       end
 
-      # There is not data to corridnate what IPs go with what networks
-      # So if there is only 1 of each link them together.
-      if result.length == 1 && guest_device_uids.length == 1
-        guest_device = guest_device_uids[guest_device_uids.keys.first]
-        guest_device[:network] = result.first unless guest_device.nil?
+      update_guest_device_network(guest_device_uids, network_by_mac)
+
+      all_networks
+    end
+
+    def get_ipaddresses(inv_net)
+      ipv4addresses = []
+      ipv6addresses = []
+
+      inv_net.to_miq_a.each do |net|
+        if net.version == "v4"
+          ipv4addresses << net.address
```

Comment:
maybe:
(net.version == "v4" ? ipv4addresses : ipv6addresses) << net.address

not sure if it is better. up to you.

Reply:
done